### ISSUE TITLE
IDISP006 on explicit interface implementations

### ISSUE THREAD

**Issue by sunsided** [DotNetAnalyzers/IDisposableAnalyzers#227]
`void IDisposable.Dispose()` should be treated as if it was `public void Dispose()` for [`IDISP006`](https://github.com/DotNetAnalyzers/IDisposableAnalyzers/blob/master/documentation/IDISP006.md) "Implement IDisposable".

Specifically, when `IDisposable.Dispose()` is implemented explicitly, the warning is raised on values that are indeed correctly disposed:

```csharp
public sealed class SomeOptions : IDisposable
{
    private readonly IDisposable _registration;

    public SomeOptions(IConfiguration section)
    {
        _registration = ChangeToken.OnChange(
            section.GetReloadToken, ApplyConfiguration, section);
    }

    // triggers the warning
    void IDisposable.Dispose()
    {
        _registration.Dispose();
    }

    private void ApplyConfiguration(IConfiguration section) {}
}
```

The following code fixed the issue, but also makes the `Dispose` method available to users that may assume it's meant to be called by them (when it isn't):

```csharp
public sealed class SomeOptions : IDisposable
{
    private readonly IDisposable _registration;

    public SomeOptions(IConfiguration section)
    {
        _registration = ChangeToken.OnChange(
            section.GetReloadToken, ApplyConfiguration, section);
    }

    // removes the warning
    public void Dispose()
    {
        _registration.Dispose();
    }

    private void ApplyConfiguration(IConfiguration section) {}
}
```

**Comment by JohanLarsson** [comment]
Thanks for reporting!

**Comment by JohanLarsson** [comment]
https://www.nuget.org/packages/IDisposableAnalyzers/3.4.0

**Comment by sunsided** [comment]
Awesome! Thank you!
